src/celeritas/field/FieldPropagator.hh

```diff
@@ -183,10 +185,17 @@ CELER_FUNCTION auto FieldPropagator<DriverT>::operator()(real_type step)
         {
             // The straight-line intersection point is a distance less than
             // `delta_intersection` from the substep's end position.
-            // Commit the proposed state's momentum but use the
-            // post-boundary-crossing track position for consistency
+            // Commit the proposed state's momentum, use the
+            // post-boundary-crossing track position for consistency, and
+            // conservatively reduce the *reported* traveled distance to avoid
+            // coincident boundary crossings.
             result.boundary = true;
-            result.distance += substep.step;
+            real_type miss_distance = detail::calc_miss_distance(
```

Comment:
The name, calc_miss_distance is somewhat confusing (as we conventionally take detail::distance_chord as the miss distance.  I still do not fully understand why we subtract this miss_distance from the substep.step here, but will try harder.

Reply:
The "miss distance" here is $|B - C|$ in the diagram I just added to the PR description. It's a conservative way of adjusting the actual substep traveled distance: for linear substeps, it's exactly the path length traveled from the start _A_ to the boundary _C_, and for more curved distances it's less accurate but (importantly!) prevents us from falsely reaching a boundary at exactly the physics step length.